qiskit/qobj/qasm_qobj.py

```diff
@@ -0,0 +1,482 @@
+        """Instatiate a QasmQobjExperiment.
+
+        Args:
+            config (QasmQobjExperimentConfig): A config object for the experiment
+            header (QasmQobjExperimentHeader): A header object for the experiment
+            instructions (list): A list of :class:`QasmQobjInstruction` objects
+        """
+        self.config = config or QasmQobjExperimentConfig()
+        self.header = header or QasmQobjExperimentHeader()
+        self.instructions = instructions or []
+
+    def to_dict(self):
+        """Return a dictionary format representation of the Experiment.
+
+        Returns:
+            dict: The dictionary form of the QasmQObjExperiment.
+        """
+        out_dict = {
+            'config': self.config.to_dict(),
+            'header': self.header.to_dict(),
+            'instructions': [x.to_dict() for x in self.instructions]
+        }
+        return out_dict
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create a new QasmQobjExperiment object from a dictionary.
+
+        Args:
+            data (dict): A dictionary for the experiment config
+
+        Returns:
+            QasmQobjExperiment: The object from the input dictionary.
+        """
+        config = None
+        if 'config' in data:
+            config = QasmQobjExperimentConfig.from_dict(data.pop('config'))
+        header = None
+        if 'header' in data:
+            header = QasmQobjExperimentHeader.from_dict(data.pop('header'))
+        instructions = None
+        if 'instructions' in data:
+            instructions = [
+                QasmQobjInstruction.from_dict(
+                    inst) for inst in data.pop('instructions')]
+        return cls(config, header, instructions)
+
+    def __eq__(self, other):
+        if isinstance(other, QasmQobjExperiment):
+            if self.to_dict() == other.to_dict():
+                return True
+        return False
+
+
+class QasmQobjConfig(SimpleNamespace):
+    """A configuration for a QASM Qobj."""
+
+    def __init__(self, shots=None, max_credits=None, seed_simulator=None,
+                 memory=None, parameter_binds=None, memory_slots=None,
+                 n_qubits=None, **kwargs):
+        """Model for RunConfig.
+
+        Args:
+            shots (int): the number of shots.
+            max_credits (int): the max_credits to use on the IBMQ public devices.
+            seed_simulator (int): the seed to use in the simulator
+            memory (bool): whether to request memory from backend (per-shot readouts)
+            parameter_binds (list[dict]): List of parameter bindings
+            memory_slots (int): The number of memory slots on the device
+            n_qubits (int): The number of qubits on the device
+            kwargs: Additional free form key value fields to add to the
+                configuration.
+        """
+        if shots is not None:
+            self.__dict__['shots'] = int(shots)
```

Comment:
Same as what @jyu00 noted above.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/3383/commits/e10bbc951178f497260a959d900a0d4fbeae895e